packages/terra-tabs/src/common-tabs/_Tabs.jsx

```diff
@@ -336,30 +385,83 @@ class Tabs extends React.Component {
 
   wrapOnSelect(onSelect) {
     return (itemKey, metaData) => {
-      this.setIsOpen(false);
-      onSelect(itemKey, metaData);
+      const updatedTabData = this.state.visibleTabData.map((tab) => ({
+        ...tab,
+        isSelected: tab.itemKey === itemKey,
+      }));
+      this.setState({ visibleTabData: updatedTabData }, () => {
+        this.setIsOpen(false);
+        onSelect(itemKey, metaData);
+      });
     };
   }
 
   wrapOnSelectHidden(onSelect) {
     return (itemKey, metaData) => {
       if (this.isOpen) {
-        onSelect(itemKey, metaData);
+        const updatedTabData = this.state.visibleTabData.map((tab) => ({
+          ...tab,
+          isSelected: tab.id === itemKey,
+        }));
+        this.setState({ visibleTabData: updatedTabData }, () => {
+          onSelect(itemKey, metaData);
+          this.setIsOpen(!this.isOpen);
+        });
       }
-      this.setIsOpen(!this.isOpen);
     };
   }
 
+  wrapOnClose() {
+    return (itemKey, event) => {
+      this.addButtonToggle = false;
+      this.setIsOpen(false);
+      let removedTabIndex = -1;
+      const updatedTabData = this.state.visibleTabData
+        .map((tab, index) => {
+          if (tab.itemKey === itemKey && tab.isSelected === true) {
+            removedTabIndex = index;
+          }
+          return { ...tab, originalIndex: index };
+        })
+        .filter((tab) => tab.itemKey !== itemKey);
+
+      if (!updatedTabData.some((tab) => tab.isSelected === true) && updatedTabData.length > 0 && removedTabIndex !== 0 && updatedTabData[removedTabIndex - 1] !== undefined && updatedTabData[removedTabIndex - 1].isDisabled !== true) {
+        if (updatedTabData[removedTabIndex - 1].isSelected !== undefined) {
+          updatedTabData[removedTabIndex - 1].isSelected = true;
+        }
+      } else if (!updatedTabData.some((tab) => tab.isSelected === true) && updatedTabData.length > 0 && removedTabIndex !== 0 && updatedTabData[removedTabIndex - 1] !== undefined && updatedTabData[removedTabIndex - 1].isDisabled === true) {
+        if (updatedTabData[removedTabIndex - 2]?.isSelected !== undefined) {
+          updatedTabData[removedTabIndex - 2].isSelected = true;
+        }
+      } else if (removedTabIndex >= 0 && removedTabIndex < updatedTabData.length) {
+        if (updatedTabData[removedTabIndex]?.isSelected !== undefined) {
+          updatedTabData[removedTabIndex].isSelected = true;
+        }
+      }
```

Comment:
Do we need this conditional workflows as onclose is meant to provide only the callback for consumer to re-organize tab data based on ID provided by the callback

Reply:
This condition have been implement to handle when the active tab got close and the previous tab is isDisabled one. So the logic should skip that tab and make the tab active which is beside it .